### ISSUE TITLE
Doesn't start (0.7.4c)

### ISSUE THREAD

**Issue by echaskaris** [00-Evan/shattered-pixel-dungeon#338]
And I can't downgrade, I get an error. I use fdroid. Thanks.

The message I get when starting the app is "Unfortunately Shattered Pixel Dungeon has stopped"

Kitkat 4.4.4

**Comment by 00-Evan** [comment]
I'll need more info than that. Could you please capture the logcat output when you attempt to launch the game? There should be a more detailed error report there.

**Comment by echaskaris** [comment]
Is this correct?
https://drive.google.com/file/d/0BwCfXrh3ceiBa1piTHo5NXpUeS1HMEZUd2U5Q3ZzanR6TW8w/view?usp=sharing

**Comment by 00-Evan** [comment]
Alright so you've got a class not found exception, possibly caused by a bugged install. I would suggesting reinstalling the game. If you don't want to uninstall and lose any save data you can manually download and re-install the 0.7.4c APK from the F-droid website here: https://f-droid.org/en/packages/com.shatteredpixel.shatteredpixeldungeon/

For reference, this was the relevant part of the log output:
```
08-23 09:09:42.887 E/AndroidRuntime(13584): FATAL EXCEPTION: main
08-23 09:09:42.887 E/AndroidRuntime(13584): Process: com.shatteredpixel.shatteredpixeldungeon, PID: 13584
08-23 09:09:42.887 E/AndroidRuntime(13584): java.lang.RuntimeException: Unable to start activity ComponentInfo{com.shatteredpixel.shatteredpixeldungeon/com.shatteredpixel.shatteredpixeldungeon.android.AndroidLauncher}: java.lang.RuntimeException: Couldn't construct AndroidInput, this should never happen
08-23 09:09:42.887 E/AndroidRuntime(13584): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2319)
08-23 09:09:42.887 E/AndroidRuntime(13584): 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2370)
08-23 09:09:42.887 E/AndroidRuntime(13584): 	at android.app.ActivityThread.access$800(ActivityThread.java:155)
08-23 09:09:42.887 E/AndroidRuntime(13584): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1243)
08-23 09:09:42.887 E/AndroidRuntime(13584): 	at android.os.Handler.dispatchMessage(Handler.java:102)
08-23 09:09:42.887 E/AndroidRuntime(13584): 	at android.os.Looper.loop(Looper.java:136)
08-23 09:09:42.887 E/AndroidRuntime(13584): 	at android.app.ActivityThread.main(ActivityThread.java:5426)
08-23 09:09:42.887 E/AndroidRuntime(13584): 	at java.lang.reflect.Method.invokeNative(Native Method)
08-23 09:09:42.887 E/AndroidRuntime(13584): 	at java.lang.reflect.Method.invoke(Method.java:515)
08-23 09:09:42.887 E/AndroidRuntime(13584): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1268)
08-23 09:09:42.887 E/AndroidRuntime(13584): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1084)
08-23 09:09:42.887 E/AndroidRuntime(13584): 	at dalvik.system.NativeStart.main(Native Method)
08-23 09:09:42.887 E/AndroidRuntime(13584): Caused by: java.lang.RuntimeException: Couldn't construct AndroidInput, this should never happen
08-23 09:09:42.887 E/AndroidRuntime(13584): 	at a.b.a.b.a.c.initialize(AndroidApplication.java:45)
08-23 09:09:42.887 E/AndroidRuntime(13584): 	at com.shatteredpixel.shatteredpixeldungeon.android.AndroidLauncher.onCreate(AndroidLauncher.java:21)
08-23 09:09:42.887 E/AndroidRuntime(13584): 	at android.app.Activity.performCreate(Activity.java:5296)
08-23 09:09:42.887 E/AndroidRuntime(13584): 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1094)
08-23 09:09:42.887 E/AndroidRuntime(13584): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2283)
08-23 09:09:42.887 E/AndroidRuntime(13584): 	... 11 more
08-23 09:09:42.887 E/AndroidRuntime(13584): Caused by: java.lang.ClassNotFoundException: a.b.a.b.a.q
08-23 09:09:42.887 E/AndroidRuntime(13584): 	at java.lang.Class.classForName(Native Method)
08-23 09:09:42.887 E/AndroidRuntime(13584): 	at java.lang.Class.forName(Class.java:251)
08-23 09:09:42.887 E/AndroidRuntime(13584): 	at java.lang.Class.forName(Class.java:216)
08-23 09:09:42.887 E/AndroidRuntime(13584): 	at a.b.a.b.a.c.initialize(AndroidApplication.java:6)
08-23 09:09:42.887 E/AndroidRuntime(13584): 	... 15 more
08-23 09:09:42.887 E/AndroidRuntime(13584): Caused by: java.lang.VerifyError: a/b/a/b/a/p
08-23 09:09:42.887 E/AndroidRuntime(13584): 	... 19 more
```

**Comment by echaskaris** [comment]
Downloaded and installed the apk, shattered still won't start.

**Comment by 00-Evan** [comment]
hmm, I can reproduce this error but only when testing with a similarly older device (android 4.1.2 in my case). My android 9.0 device runs the f-droid APK just fine. Furthermore any builds I make using the same code work correctly. This looks like some sort of issue either with F-droid's build process, or some error in my proguard configuration which F-droid is exposing.

I'll make an issue on F-droid's repository tommorow and see if anyone where has a better idea why something like this might happen. In the meantime I'm afraid the only way to downgrade is to uninstall the app and lose your data (though you should be able to use ADB backup and restore to preserve badges/rankings)

**Comment by echaskaris** [comment]
I don't know if it's relevant but I had an update to fdroid as well as shattered.

**Comment by sviscapi** [comment]
Hi all,

Same bug here, but on a rather "strange" plateform:

Sony Xperia X with F-Droid 1.7.1 running on Jolla's Sailfish OS 3.1.0.11 (+ Android 4.4 compatibility layer).

https://jolla.com/
https://jolla.com/sailfishx/
https://sailfishos.org/

I had to uninstall Shattered Pixel Dungeon and lost all my badges and rankings in the process (no big deal though). Shattered Pixel Dungeon 0.7.4 works flawlessly.

Cheers,

Sam a.k.a "Magic Sam" from the Open Pandora community

https://pyra-handheld.com/boards/threads/shattered-pixel-dungeon.83410/
https://repo.openpandora.org/?page=detail&app=shatteredpixeldungeon_ptitseb

P.S.: @00-Evan and @watabou : thank you both very much for this amazing game !

**Comment by 00-Evan** [comment]
F-Droid Issue opened here: https://gitlab.com/fdroid/fdroiddata/issues/1759

**Comment by 00-Evan** [comment]
Fixed in this commit: bb9a1850d87d764ffb3861c4d61e17970daeff11

Will make a merge request on F-droid and hopefully 0.7.4d will show up there shortly with this working correctly.
